app/gui/view/debug_scene/cloud-dashboard/src/projects_table.rs

```diff
@@ -0,0 +1,698 @@
+        /// The first row in the section the requested visible entry is in.
+        ///
+        /// This is always `0` because we only have one section, so the first row is the first
+        /// visible one in the table.
+        const SECTION_START: usize = 0;
+        /// The last row in the section the requested visible entry is in.
+        ///
+        /// This is always [`usize::MAX`] because we only have one section, so all rows are in the
+        /// current section.
+        const SECTION_END: usize = usize::MAX;
+
+        let Position { row: _, column } = position;
+        let position = (SECTION_START, column).into();
+        let model = self.header_entry_model(position);
+        let section_range = SECTION_START..SECTION_END;
+        (section_range, column, model)
+    }
+
+    fn header_entry_model(&self, position: Position) -> EntryModel {
+        let Position { row, column: _ } = position;
+        assert!(row == 0, "Header row was {row}, but it is expected to be first row in the table.");
+        let column = column_for_entry(position);
+        let entry_model = column.map(|column| EntryModel {
+            text:           column.to_string().into(),
+            disabled:       Immutable(true),
+            override_width: Immutable(None),
+        });
+        let entry_model = entry_model.unwrap_or_else(invalid_entry_model);
+        entry_model
+    }
+}
+
+
+// === Setter `impl` ===
+
+impl Model {
+    fn set_projects(&self, projects: Rc<Vec<view::project::Project>>) {
+        *self.projects.raw.borrow_mut() = projects.to_vec();
+
+        let row = self.rows();
+        let col = Columns::LEN;
+        self.projects_table.resize_grid(row, col);
+    }
+}
+
+/// Returns an [`EntryModel`] representing a [`Project`] in the [`ProjectsTable`].
+///
+/// [`Project`]: ::enso_cloud_view::project::Project
+fn project_entry_model(project: &view::project::Project, column: Columns) -> EntryModel {
+    // Map the requested column to the corresponding field of the `Project` struct.
+    let model = match column {
+        Columns::Projects => {
+            // TODO [NP]: https://www.pivotaltracker.com/story/show/183909391
+            // `"Home Screen" User` can see an `Icon` representing the `Project`'s state.
+            let state = &project.state;
+            let name = &project.name;
+            let state = match state {
+                view::project::StateTag::New => "New".to_string(),
+                view::project::StateTag::Created => "Created".to_string(),
+                view::project::StateTag::OpenInProgress => "OpenInProgress".to_string(),
+                view::project::StateTag::Opened => "Opened".to_string(),
+                view::project::StateTag::Closed => "Closed".to_string(),
+            };
+            format!("({state}) {name}")
+        }
+        Columns::LastModified => LAST_MODIFIED.to_string(),
+        Columns::SharedWith => SHARED_WITH.to_string(),
+        // TODO [NP]: 003
```

Comment:
This is a strange TODO - please check all TODOs and check if they have a valid form